src/updater/releaser.rs

```diff
@@ -0,0 +1,215 @@
+                .map_err(|e| io::Error::new(io::ErrorKind::Other, e.to_string()))?;
+            if latest.tag_name.starts_with('v') {
+                latest.tag_name = latest.tag_name.as_str()[1..].to_string();
+            }
+            *self.latest_release.borrow_mut() = Some(latest);
+            Ok(())
+        }
+    }
+}
+
+impl Releaser for GithubReleaser {
+    fn new<S: Into<String>>(repo_name: S) -> GithubReleaser {
+        GithubReleaser::new(repo_name)
+    }
+
+    // This implementation of Releaser will favor urls that end with `alfred3workflow`
+    // over `alfredworkflow`
+    fn downloadable_url(&self) -> Result<String, io::Error> {
+        let release = self.latest_release.borrow();
+        let urls = release.as_ref().map(|r| {
+            r.assets
+                .iter()
+                .filter(|asset| {
+                    asset.state == "uploaded"
+                        && (asset.browser_download_url.ends_with("alfredworkflow")
+                            || asset.browser_download_url.ends_with("alfred3workflow"))
+                })
+                .map(|asset| &asset.browser_download_url)
+                .collect::<Vec<&String>>()
+        });
+        let urls = urls.ok_or_else(|| {
+            io::Error::new(
+                io::ErrorKind::Other,
+                "no release item available, did you check with newer_than?",
+            )
+        })?;
+
+        if urls.len() == 1 {
+            Ok(urls[0].clone())
+        } else if urls.len() > 1 {
+            let url = urls.iter().find(|item| item.ends_with("alfred3workflow"));
+            let u = match url {
+                Some(&link) => (*link).clone(),
+                None => urls[0].clone(),
+            };
+            Ok(u)
+        } else {
+            Err(io::Error::new(
+                io::ErrorKind::Other,
+                "no usable download url",
+            ))
+        }
+    }
+
+    fn newer_than(&self, v: &Version) -> Result<bool, io::Error> {
+        if self.latest_release.borrow().is_none() {
+            self.latest_release_data()?;
+        }
+
+        let latest_version = self.latest_release
+            .borrow()
+            .as_ref()
+            .map(|r| Version::parse(&r.tag_name).ok())
+            .ok_or_else(|| {
+                io::Error::new(io::ErrorKind::Other, "Couldn't parse fetched version.")
+            })?;
+        Ok(*v < latest_version.ok_or_else(|| str_to_io_err("should have version at this point"))?)
+    }
+}
+
+pub fn str_to_io_err<S: AsRef<str>>(s: S) -> io::Error {
+    io::Error::new(io::ErrorKind::Other, s.as_ref())
+}
+
+#[cfg(test)]
+pub mod tests {
+    use super::*;
+    use mockito::mock;
+    use mockito::Matcher;
+    use mockito::Mock;
+
+    #[test]
+    fn test_get_latest_release() {
+        let releaser = GithubReleaser::new("spamwax/alfred-pinboard-rs");
+
+        assert!(
+            releaser
+                .newer_than(&Version::from((0, 0, 0)))
```

Comment:
As with the other tests, I'm uncomfortable having this do actual network access. Can you mock out the network call?